core/server/models/plugins/has-posts.js

```diff
@@ -0,0 +1,61 @@
+const _ = require('lodash');
+const _debug = require('ghost-ignition').debug._base;
+const debug = _debug('ghost-query');
+
+const addHasPostsWhere = function () {
+    const tableName = _.result(this, 'tableName');
+    const comparisonField = `${tableName}.id`;
+
+    this.query(function (qb) {
+        return qb.whereIn(comparisonField, function () {
+            const innerQb = this
+                .distinct('posts_authors.author_id')
```

Comment:
I assume this logic will become dynamic to handle authors and tags.

Reply:
Yeap, similar to how relations are configured :)